### PR TITLE
[misc] Fix build warnings

### PR DESCRIPTION
**Pull request by songqing**
## Description

Fix build warnings of cpp code.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [*] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [*] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [*] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [*] All changes have test coverage
- [*] Code is well-documented
- [*] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

**Comment by yaox12** [comment]
@dgl-bot

File: src/graph/heterograph_capi.cc [hunk 1]
```diff
@@ -404,6 +404,7 @@ DGL_REGISTER_GLOBAL("heterograph_index._CAPI_DGLHeteroVertexSubgraph")
       HeteroGraphRef hg = args[0];
       List<Value> vids = args[1];
       bool relabel_nodes = args[2];
+      (void)relabel_nodes;
```
Comment (frozenbugs) [inline src/graph/heterograph_capi.cc:407]: why not remove L406?

Comment (songqing) [reply src/graph/heterograph_capi.cc:407]: Removing the param 'relabel_nodes' may have compatibility problem, and maybe the param is not implemented yet, I'll confirm it.

Comment (caojy1998) [reply src/graph/heterograph_capi.cc:407]: In past version, we have a check here to confirm relabel_nodes is set to True. However, now we remove this check and enables the function when relabel_nodes is False. 
So it is OK to remove this param here.

Comment (songqing) [reply src/graph/heterograph_capi.cc]: OK, the param relabel_nodes has been removed

**Comment by frozenbugs** [comment]
@dgl-bot

**Review by frozenbugs — APPROVED** [review]

**Pull request merged**